src/vs/workbench/browser/web.main.ts

```diff
@@ -0,0 +1,623 @@
+					async run(accessor: ServicesAccessor): Promise<void> {
+						const dialogService = accessor.get(IDialogService);
+						const hostService = accessor.get(IHostService);
+						const result = await dialogService.confirm({
+							message: localize(
+								'reset user data message',
+								'Would you like to reset your data (settings, keybindings, extensions, snippets and UI State) and reload?'
+							),
+						});
+						if (result.confirmed) {
+							await indexedDBUserDataProvider!.reset();
+						}
+						hostService.reload();
+					}
+				}
+			);
+		}
+
+		fileService.registerProvider(
+			Schemas.userData,
+			indexedDBUserDataProvider || new InMemoryFileSystemProvider()
+		);
+	}
+
+	private async createStorageService(
+		payload: IWorkspaceInitializationPayload,
+		environmentService: IWorkbenchEnvironmentService,
+		fileService: IFileService,
+		logService: ILogService
+	): Promise<BrowserStorageService> {
+		const storageService = new BrowserStorageService(
+			environmentService,
+			fileService
+		);
+
+		try {
+			await storageService.initialize(payload);
+
+			return storageService;
+		} catch (error) {
+			onUnexpectedError(error);
+			logService.error(error);
+
+			return storageService;
+		}
+	}
+
+	private async createWorkspaceService(
+		payload: IWorkspaceInitializationPayload,
+		environmentService: IWorkbenchEnvironmentService,
+		fileService: FileService,
+		remoteAgentService: IRemoteAgentService,
+		uriIdentityService: IUriIdentityService,
+		logService: ILogService
+	): Promise<WorkspaceService> {
+		const workspaceService = new WorkspaceService(
+			{
+				remoteAuthority: this.configuration.remoteAuthority,
+				configurationCache: new ConfigurationCache(),
+			},
+			environmentService,
+			fileService,
+			remoteAgentService,
+			uriIdentityService,
+			logService
+		);
+
+		try {
+			await workspaceService.initialize(payload);
+
+			return workspaceService;
+		} catch (error) {
+			onUnexpectedError(error);
+			logService.error(error);
+
+			return workspaceService;
+		}
+	}
+
+	private async resolveWorkspaceInitializationPayload(): Promise<IWorkspaceInitializationPayload> {
+		let workspace: IWorkspace | undefined = undefined;
+		if (this.configuration.workspaceProvider) {
+			workspace = this.configuration.workspaceProvider.workspace;
+		}
+
+		// Multi-root workspace
+		if (workspace && isWorkspaceToOpen(workspace)) {
+			return getWorkspaceIdentifier(workspace.workspaceUri);
+		}
+
+		// Single-folder workspace
+		if (workspace && isFolderToOpen(workspace)) {
+			// modify-by-vscode, use string `owner+repo:path` to generate id
+			// In this way, the workspaceState will be isolated between from different repo
+			const [owner = 'conwnet', repo = 'github1s'] = URI.parse(
+				window.location.href
+			)
+				.path.split('/')
+				.filter(Boolean);
+			const id = hash(
+				`${owner}+${repo}:${workspace.folderUri.toString()}`
+			).toString(16);
+			return { id, folder: workspace.folderUri };
+		}
```

Comment:
Should we use some comments around the code block we changed to indicate the updated parts from the original VS Code?
Something like:

```
                 // below codes are changed by github1s
		if (workspace && isFolderToOpen(workspace)) {
			// modify-by-vscode, use string `owner+repo:path` to generate id
			// In this way, the workspaceState will be isolated between from different repo
			const [owner = 'conwnet', repo = 'github1s'] = URI.parse(
				window.location.href
			)
				.path.split('/')
				.filter(Boolean);
			const id = hash(
				`${owner}+${repo}:${workspace.folderUri.toString()}`
			).toString(16);
			return { id, folder: workspace.folderUri };
		}
                 // above codes are changed by github1s
```